src/o365/graph/commands/teams/teams-channel-message-list.ts

```diff
@@ -0,0 +1,108 @@
+import auth from '../../GraphAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import { GraphItemsListCommand } from '../GraphItemsListCommand';
+import Utils from '../../../../Utils';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  teamId: string;
+  channelId: string;
+}
+
+class GraphTeamsChannelMessageListCommand extends GraphItemsListCommand<any> {
+  public get name(): string {
+    return `${commands.TEAMS_CHANNEL_MESSAGE_LIST}`;
+  }
+
+  public get description(): string {
+    return 'Lists all the messages from a channel in a Microsoft Teams team';
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    let endpoint: string = `${auth.service.resource}/beta/teams/${args.options.teamId}/channels/${args.options.channelId}/messages`;
+
+    this
+    .getAllItems(endpoint, cmd, true)
+    .then((): void => {
+        cmd.log(this.items);
```

Comment:
This output is pretty useless in text mode, because it doesn't fit on the screen and contains complex objects that are listed as `[object]`. In text mode, we should include only the essential information. All data should be available in JSON output mode when it can be processed accordingly, like we do in other commands.

Reply:
Thanks @waldekmastykarz, just looked into the other code. I will remember this next time.